include/seqan3/range/views/minimiser.hpp

```diff
@@ -0,0 +1,476 @@
+                                                 std::forward_iterator_tag,
+                                                 std::input_iterator_tag>;
+    //!\brief Tag this class depending on which concept `it_t` models.
+    using iterator_concept = iterator_category;
+    //!\}
+
+    /*!\name Constructors, destructor and assignment
+     * \{
+     */
+    constexpr window_iterator()                                    = default; //!< Defaulted.
+    constexpr window_iterator(window_iterator const &)             = default; //!< Defaulted.
+    constexpr window_iterator(window_iterator &&)                  = default; //!< Defaulted.
+    constexpr window_iterator & operator=(window_iterator const &) = default; //!< Defaulted.
+    constexpr window_iterator & operator=(window_iterator &&)      = default; //!< Defaulted.
+    ~window_iterator()                                             = default; //!< Defaulted.
+
+    /*!\brief                              Construct from begin and end iterators of a given range over
+    *                                      std::totally_ordered values, and the number of values per window.
+    * /param[in] it_start                  Iterator pointing to the first position of the std::totally_ordered range.
+    * /param[in] it_end                    Iterator pointing to the last position of the std::totally_ordered range.
+    * /param[in] window_values_size        The number of values in one window.
+    *
+    * \details
+    *
+    * Looks at the number of values per window, returns the smallest as minimiser and shifts then by one to repeat
+    * this action. If a minimiser in consecutive windows is the same, it is returned only once.
+    *
+    */
+    window_iterator(it_t it_start, sentinel_t it_end, uint32_t window_values_size) :
+                    urange_end{it_end}, window_right{it_start}
+    {
+        if (window_values_size > std::ranges::distance(window_right, urange_end))
+            window_values_size = std::ranges::distance(window_right, urange_end);
+        if (window_right != urange_end)
+            window_first(window_values_size);
+    }
+    //!\}
+
+    //!\anchor window_iterator_comparison
+    //!\name Comparison operators
+    //!\{
+
+    //!\brief Compare to the sentinel of the underlying range.
+    friend bool operator==(window_iterator const & lhs, sentinel_t const & rhs) noexcept
+    {
+        return lhs.window_right == rhs;
+    }
+
+    //!\brief Compare to iterator on underlying range.
+    friend bool operator==(sentinel_t const & lhs, window_iterator const & rhs) noexcept
+    {
+        return lhs == rhs.window_right;
+    }
+
+    //!\brief Compare to the sentinel of the underlying range.
+    friend bool operator==(window_iterator const & lhs, window_iterator const & rhs) noexcept
+    {
+        return std::tuple(lhs.window_right, lhs.window_values.size()) ==
+               std::tuple(rhs.window_right, rhs.window_values.size());
+    }
```

Comment:
The reason why I changed it to tuple was that it got rid of the error "cannot bind non-const lvalue reference of type ‘long unsigned int&’ to an rvalue of type ‘std::deque<long unsigned int>::size_type’ {aka ‘long unsigned int’}"

Reply:
ah yes, that because .size() returns a temporary and cannot be bound by std::tie.
Then maybe just do it like in the operators below
```suggestion
    friend bool operator==(window_iterator const & lhs, window_iterator const & rhs) noexcept
    {
        return lhs.window_right == rhs.window_right &&
               lhs.window_values.size() == rhs.window_values.size();
    }
```